aten/src/ATen/native/Sorting.cpp

```diff
@@ -271,8 +267,124 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
   return std::forward_as_tuple(values, indices);
 }
 
+std::tuple<Tensor&, Tensor&> median_out_impl(
+    Tensor& values,
+    Tensor& indices,
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim,
+    bool ignore_nan) {
+  NoNamesGuard guard;
+
+  dim = at::maybe_wrap_dim(dim, self.dim());
+  TORCH_CHECK(self.numel() > 0, "median() input tensor cannot be empty");
```

Comment:
median of a tensor with no elements returns nan in NumPy - should we do the same?

Reply:
The median of an empty set is undefined. I think we should have a standard policy for undefined reductions. Either we return NaN or we raise an error. NumPy is not consistent with this, np.min, np.max and np.quantile raise errors while no.median and np.mean return NaN and only raise a warning. But these are actually downstream erros.

Reply:
Here's Francisco's suggestion, which seems reasonable: https://github.com/pytorch/pytorch/issues/34907.

What are your thoughts?

Reply:
That makes sense, I'll update the code to follow this new standard.